File: GuacUtil.js
// Credit: https://github.com/CollabVM-Unofficial/yanderebot
function decodeCommand(cypher) {
  var sections = [];
  var bump = 0;
  while (sections.length <= 50 && cypher.length >= bump) {
    var current = cypher.substring(bump);
    var length = parseInt(current.substring(current.search(/\./) - 2));
    var paramater = current.substring(
      length.toString().length + 1,
      Math.floor(length / 10) + 2 + length
    );
    sections[sections.length] = paramater;
    bump += Math.floor(length / 10) + 3 + length;
  }
  sections[sections.length - 1] = sections[sections.length - 1].substring(
    0,
    sections[sections.length - 1].length - 1
  );
  return sections;
}

function encodeCommand(cypher) {
  var command = "";
  for (var i = 0; i < cypher.length; i++) {
    var current = cypher[i];
    command += current.length + "." + current;
    command += i < cypher.length - 1 ? "," : ";";
  }
  return command;
}

File: GuacUtil.java
// Credit: Some guest on CollabVM who posted this to pastebin.
import java.util.*;

class GuacUtil {
    static String EncodeGuac(final String... cypher) {
        final StringBuilder command = new StringBuilder();
        for (int i = 0; i < cypher.length; i++) {
            final String current = cypher[i];
            command.append(current.length());
            command.append('.');
            command.append(current);
            command.append(i < cypher.length - 1 ? ',' : ';');
        }

        return command.toString();
    }

    static String[] DecodeGuac(final String str) {
        int pos = -1;
        final ArrayList<String> sections = new ArrayList<>();

        for (;;) {
            final int len = str.indexOf('.', pos + 1);
            if (len == -1) {
                break;
            }

            pos = Integer.parseInt(str.substring(pos + 1, pos + 1 + len - (pos + 1))) + len + 1;
            String repl = str.substring(len + 1, len + 1 + pos - (len + 1));
            repl = repl.replaceAll("&#x27;", "'").replaceAll("&quot;", "\"").replaceAll("&#x2F;", "/")
                    .replaceAll("&lt;", "<").replaceAll("&gt;", ">").replaceAll("&amp;", "&");
            sections.add(repl);

            if (str.substring(pos, pos + 1).equals(";")) {
                break;
            }
        }

        return sections.toArray(new String[0]);
    }
}